File: in_Java/0806 Number of Lines To Write String.java
class Solution {
    public int[] numberOfLines(int[] widths, String S) {
        int[] res = new int[2];
        if(S == null || S.length() == 0) return res;
        int curlen = 0;
        int lines = 1;
        for(char c : S.toCharArray()){
            int idx = widths[c - 'a'];
            if(curlen + idx <= 100){
                curlen += idx;
            }else{
                curlen = idx;
                lines++;
            }
        }
        res[0] = lines;
        res[1] = curlen;
        return res;
    }
}

File: in_Python/0806 Number of Lines To Write String.py
class Solution(object):
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        res = 1
        cur = 0
        for s in S:
            w = widths[ord(s) - ord('a')]
            if cur + w <= 100:
                cur += w
            else:
                res += 1
                cur = w
        return [res, cur]